hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -752,22 +752,14 @@ public PendingKeysDeletion getPendingDeletionKeys(
 
             // Skip the key if the filter doesn't allow the file to be deleted.
             if (filter == null || filter.apply(Table.newKeyValue(kv.getKey(), info))) {
-              BlockGroup.Builder keyBlocksBuilder = BlockGroup.newBuilder().setKeyName(kv.getKey());
-              if (isDataDistributionEnabled) {
-                List<DeletedBlock> deletedBlocks = info.getKeyLocationVersions().stream()
-                    .flatMap(versionLocations -> versionLocations.getLocationList().stream()
-                        .map(b -> new DeletedBlock(new BlockID(b.getContainerID(), b.getLocalID()),
-                            b.getLength(), QuotaUtil.getReplicatedSize(b.getLength(), info.getReplicationConfig()))))
-                    .collect(Collectors.toList());
-                keyBlocksBuilder.addAllDeletedBlocks(deletedBlocks);
-              } else {
-                List<BlockID> blockIDS = info.getKeyLocationVersions().stream()
-                    .flatMap(versionLocations -> versionLocations.getLocationList().stream()
-                        .map(b -> new BlockID(b.getContainerID(), b.getLocalID()))).collect(Collectors.toList());
-                keyBlocksBuilder.addAllBlockIDs(blockIDS);
-              }
-              BlockGroup keyBlocks = keyBlocksBuilder.build();
-              int keyBlockSerializedSize = keyBlocks.getProto().getSerializedSize();
+              List<DeletedBlock> deletedBlocks = info.getKeyLocationVersions().stream()
+                  .flatMap(versionLocations -> versionLocations.getLocationList().stream()
+                      .map(b -> new DeletedBlock(new BlockID(b.getContainerID(), b.getLocalID()),
+                          b.getLength(), QuotaUtil.getReplicatedSize(b.getLength(), info.getReplicationConfig()))))
+                  .collect(Collectors.toList());
+              BlockGroup keyBlocks = BlockGroup.newBuilder().setKeyName(kv.getKey())
+                  .addAllDeletedBlocks(deletedBlocks).build();
+              int keyBlockSerializedSize = keyBlocks.getProto(isDataDistributionEnabled).getSerializedSize();
```

Comment:
we can have getProto() and its implementation, have default as true for this as new SCM, will not consider scm version for this as assumed to be new. and no impact.